io/src/main/resources/laika/helium/css/nav.css

```
@@ -346,6 +346,8 @@ ul.nav-list, #page-nav ul {
 #page-nav {
   display: none;
   position: fixed;
+  height: 80%;
```

```suggestion
  max-height: calc(100% - 90px - var(--top-bar-height));
```

The problem with the suggested solution is that it fixes the edge cases while breaking the most common cases: all pages with just a few entries in page navigation now get a long box with a lot of empty space which looks broken.

Using `max-height` instead of `height` fixes part of the problem. But it still keeps the issue you observed where it can start to look strange with very small page heights, because it does not preserve a fixed margin below the navigation box. The suggested change also improves on that: by using 100% and subtracting the height of the top navigation and 90 pixel for the space above and below the navigation box, we force a fixed margin on the bottom.

Would be great if you'd also quickly test the suggested change before accepting it.

---

Thank you for the refinement, those look great!